lib/HunkView.ml

```diff
@@ -67,91 +89,197 @@ let ui_hunk_summary (hunk : string Patch.hunk) : Nottui.ui =
       at_symbols;
     ]
 
-let ui_unified_diff (hunk : string Patch.hunk) : Nottui.ui =
-  let hunk_summary = ui_hunk_summary hunk in
-  let hunk_content =
-    let blocks = Block.of_hunk hunk.Patch.lines in
-    let single_line_changes =
-      List.for_all
-        (function
-          | Block.Changed { mine; their; _ } ->
-              List.length mine = 1 && List.length their = 1
-          | _ -> true)
-        blocks
-    in
-    if single_line_changes then
-      let word_diff_blocks = List.map WordDiff.compute blocks in
-      let word_diff_lines = Block.to_hunk word_diff_blocks in
-      WordDiff.render_hunk_lines word_diff_lines
-    else WordDiff.render_hunk hunk
+let render_line_number (mine_num : int) (their_num : int)
+    (diff_type : [ `Added | `Removed | `Unchanged ]) (mode : rendering_mode) :
+    Ui.t =
+  let attr_option =
```

Comment:
An other optional argument tips: If you name this `attr`, you could pass it as `?attr` instead of `?attr:attr`.

I think this is applicable here as the longer name is not useful.